static/js/addReviewClient.js

```diff
@@ -1,35 +1,89 @@
 const addReviewButton = document.querySelector(".add-review-button");
-let addReviewContent = document.querySelector(".add-review-content");
+const addReviewContent = document.querySelector(".add-review-content");
+const loggedOutEl = document.querySelector(".add-review-logged-out");
+const loggedInEl = document.querySelector(".add-review-logged-in");
 
 addReviewButton.addEventListener("click", () => {
   addReviewButton.classList.toggle("clicked");
   addReviewContent.classList.toggle("show");
 });
 
+let loggedIn = false;
+let sendReviewToken;
+
+document.querySelector("#login").addEventListener("click", async () => {
+
+  let username = document.querySelector("#username").value;
+  let password = document.querySelector("#password").value;
+  const bufferedInput = btoa(username + ":" + password);
+  try {
+    const response = await fetch(
+    new Request("/api/token", {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        "Authentication": 
+        "Basic " + bufferedInput,
+      },
+    }));
+    if (response.status === 401) {
+      alert("Fel lösenord eller användarnamn");
+      throw new Error("Invalid username or password");
+    }
+    loggedIn = true;
+    loggedInEl.classList.add("show");
+    loggedInEl.classList.remove("hide");
+    loggedOutEl.classList.add("hide");
+
+    const data = await response.json();
+    sendReviewToken = data.token;
+
+  } catch (e) {
+    console.log(e.message);
+  }
+});
+
+function triggerNotLoggedInMessage() {
+  if (!loggedIn) {
+    const notLogged = document.querySelector(".add-review-not-logged-warning")
+    notLogged.classList.remove("hide");
+    notLogged.classList.add("show");
+    
+    setTimeout(() => {
+      notLogged.classList.remove("show");
+      notLogged.classList.add("hide");
+    }, 1500)
+    return false;
+  } else {
+    return true;
+  }
+}
+
 document
   .querySelector(".submit-review")
   .addEventListener("click", async (ev) => {
     ev.preventDefault();
+
+    const isLoggedIn = triggerNotLoggedInMessage();
```

Comment:
I see what you're doing here, and it's not by any means wrong, but it can be confusing to use functions and their return values like this. Functions usually DO something, and they RETURN something. Often you design functions to either have as their main purpose to DO something (and return nothing) or to have as their main purpose to calculate and RETURN something (and then DO nothing, in the sense that they have no side effects).

What you've done here is create a function that does something AND returns something. That's fine, but then you have to be extra careful that the two things harmonize. To me, they don't, in this case.

The function is called `triggerNotLoggedInMessage()` which tells me that what it will do is to trigger something when the user is NOT logged in.

The function returns a boolean. When should I expect it to be true?

I think it's equally reasonable to assume that the boolean will be `true` if the error message was triggered (because that's what the function is said to DO), as it is to assume that the boolean will be `true` when the user is logged in (because that's a positive case).

Personally, I think I would have split this up into two functions. One called `userIsLoggedIn()` (which will clearly return `true` if the user is logged in) and another which is called `triggerNotLoggedInMessage()` which I would only invoke if the first one returns `false`, and ignore any return value from.